tests/sagemaker-tests/test_sagemaker_utils.py

```diff
@@ -221,3 +233,156 @@ def test_wrap_model_artifacts():
+    hyphons, letters, or digits. And the string must end with a letter or digit.
+    """
+    # Test case 1: normal case, following the regex, including all three parts and all match
+    valid_names = ['ab-cd-9',  'abc', 'a-b', 'A1-foo2', 'Z9', 'x-y-z', 'abc123']
+    for valid_name in valid_names:
+        assert check_name_format(valid_name) == valid_name
+
+    # Test case 2: abnormal cases
+    invalid_names = [
+                "-abc",     # starts with hyphen
+                "abc-",     # ends with hyphen
+                "a--",      # ends with hyphen
+                "a_",       # contains invalid character
+                "a",        # too short to match ([...]*[a-zA-Z0-9])
+                ""         # empty string
+            ]
+    for invalid_name in invalid_names:
+        with pytest.raises(ArgumentTypeError, match='failed to satisfy regular expression pattern'):
+            check_name_format(invalid_name)
+
+@patch('launch_utils.boto3.client')
+def test_check_tarfile_s3_object(mock_boto_client):
+    """ The the check if tarfile object in S3 url correct and exist
+    """
+    mock_s3 = Mock()
+    mock_boto_client.return_value = mock_s3
+
+    # Test case 1: normal case. S3 url is right and the object ends with '.tar.gz' 
+    mock_s3.head_object.return_value = "s3://graphstorm-demos/test/model.tar.gz"
+    assert check_tarfile_s3_object("s3://graphstorm-demos/test/model.tar.gz") is True
+
+    # Test case 2: abnormal cases
+    #       2.1: S3 url is incorrect, not starting with s3 or https
+    error_response = {'Error': {'Code': '404'}}
+    mock_s3.head_object.side_effect = ClientError(error_response, 'HeadObject')
+    with pytest.raises(AssertionError, match='Incorrect S3'):
+        check_tarfile_s3_object('/graphstorm-demos/test/GSF-demo.pptx')
+
+    #       2.2: S3 url is correct, but not ending with .tar.gz
+    error_response = {'Error': {'Code': '404'}}
+    mock_s3.head_object.side_effect = ClientError(error_response, 'HeadObject')
+    with pytest.raises(AssertionError, match='not a compressed tar file'):
+        check_tarfile_s3_object('s3://graphstorm-demos/test/GSF-demo.pptx')
+
+@patch('launch_utils.S3Uploader.upload')
+def test_upload_data_to_s3(mock_s3uploader):
+    """ Test the upload data to S3 function.
+    """
+    mock_s3uploader.return_value = 's3://graphstorm-demos/test/model.tar.gz'
+    
+    # Test case 1: mock successful upload
+    ret = upload_data_to_s3('s3://graphstorm-demos/test/', './model.tar.gz', 'session')
+    mock_s3uploader.assert_called_once_with('./model.tar.gz', 's3://graphstorm-demos/test/',
+                                            sagemaker_session='session')
+
+    assert ret == 's3://graphstorm-demos/test/model.tar.gz'
+
+    # Test case 2: mock unsucessful upload
+    ret = upload_data_to_s3('s3://graphstorm-demos/test/', './model.tar.gz', 'session')
+    with pytest.raises(AssertionError):
```

Comment:
Shouldn't we expect L324 will raise the error?

Reply:
No, this is a mock operation in which the `upload()` function is taken by the Mock and will return preset `.return_value` above.